examples/intrinsics/demo-hallucination.pdl

```diff
@@ -1,11 +1,86 @@
 # Granite.runtime Intrinsics Demo with PDL
 
-# Provide document(s) to Granite3-rag model with hallucination_tag
-# Granite3-rag model: Base model granite3-dense:8b setup with Granite RAG LoRA (Low-Rank Adaption) on ollama
+# Provide document(s) to granite3-rag model with hallucination_tag
+# granite3-rag model: Base model granite3-dense:8b setup with Granite RAG LoRA (Low-Rank Adaption) on Ollama
 
-# The model responds to a query providing a hallucination score
+# The model responds to a query and provides a hallucination score
 
 text:
+
+# Builds granite3-rag:8b
+- contribute: []
+  lang: python
+  code: |
+    import os
+    import shutil
+    import subprocess
+    import huggingface_hub
+    import ollama
+
+    granite3_rag_model = 'granite3-rag:8b'
+
+    # Check if rag model exists
+    found = False
+    result = ""
+    local_ollama_models = ollama.list().models
+    for model in local_ollama_models:
+        if model.model == granite3_rag_model:
+            found = True
+            result = f"Model {granite3_rag_model} found in Ollama, skipping build"
+            break
+
+    if not found:
+        # Install dependencies
+        # Define the repository and directory
+        llama_cpp_repo = "https://github.com/ggerganov/llama.cpp.git"
+        llama_cpp_dir_name = "llama.cpp"
+
+        # Clone the repository if it doesn't already exist
+        if not os.path.isdir(llama_cpp_dir_name):
+            subprocess.run(["git", "clone", "--depth", "1", llama_cpp_repo], check=True)
+
+        # Install the required packages
+        subprocess.run([
+            "pip", "install",
+            "git+https://github.com/ibm-granite-community/utils.git",
+            "huggingface_hub",
+            "langchain_community",
+            "langchain_ollama",
+            "langchain-milvus",
+            "docling",
+            "-r", f"{llama_cpp_dir_name}/requirements.txt"
+        ], check=True)
```

Comment:
Not sure if this is the best approach to install dependencies. I didn't use the `lang: command` because the `git clone` action will create a repository, and so we need additional logic such as checking if the repo is already cloned. I feel like the command can get overly complicated. Also, I wonder if installing dependencies via code blocks violates PDL design since it should be handled outside the program itself. Also, `lang: command` code blocks can create side effects that isn't obvious. For example, `git clone` creates a new repo. How do we handle this? Thoughts?

Reply:
shouldn't we add a `requirements:` field to `code` blocks? then the interpreter does this part? because... its'more complex, there are venv issues, for example.

Reply:
It raises the question again of switching to IPython and to do that like in notebooks.